tests/pipeline/test_term.py

```diff
@@ -330,6 +334,38 @@ class SomeFactorNoDType(SomeFactor):
         with self.assertRaises(InvalidDType):
             SomeFactor(dtype=1)
 
+    def test_latest_on_different_dtypes(self):
+        factor_dtypes = (int64_dtype, float64_dtype, datetime64ns_dtype)
+        for column in TestingDataSet.columns:
+            if column.dtype == bool_dtype:
+                self.assertIsInstance(column.latest, Filter)
+            elif column.dtype in factor_dtypes:
+                self.assertIsInstance(column.latest, Factor)
+            else:
+                self.fail(
+                    "Unknown dtype %s for column %s" % (column.dtype, column)
+                )
+            # These should be the same value, plus this has the convenient
+            # property of correctly handling `NaN`.
+            self.assertIs(column.missing_value, column.latest.missing_value)
+
+    def test_failure_timing_on_bad_missing_values(self):
```

Comment:
@llllllllll I think this addresses the concern you raised in person today.

I originally had made Column fail eagerly, but that was annoying because the error couldn't tell you the name of the column that failed. The compromise here is to fail in _BoundColumnDescr, which means we still barf in class construction, but we can give you an error that includes the name of the bad column.